frontend/src/components/auth/ResetPassword.tsx

```diff
@@ -2,11 +2,35 @@ import React, { useContext } from "react";
 import authAPIClient from "../../APIClients/AuthAPIClient";
 import AuthContext from "../../contexts/AuthContext";
 
-const ResetPassword = () => {
+interface ResetPasswordProps {
+  email?: string;
+}
+
+const ResetPassword = ({ email }: ResetPasswordProps) => {
   const { authenticatedUser } = useContext(AuthContext);
 
+  const isRealEmailAvailable = (): boolean => {
```

Comment:
nit: I suggest using an `email` parameter here and passing in the email to the function.  Makes the function's purpose crystal clear. 
i.e ` const isRealEmailAvailable = (email: string): boolean => .....`